### PR TITLE
Use : as delimiter for layout flag

### PR DESCRIPTION
**Pull request by ivikhrev**
https://github.com/openvinotoolkit/open_model_zoo/pull/3104#discussion_r802896207

### CODE REVIEW

**Comment by vladimir-dudnik** [comment]
@ivikhrev would be good to add some details on new format to corresponding demo descriptions, which some challenging examples, like `--layout image_info:0:?c` if this is what you meant by this change

**Comment by ivikhrev** [comment]
@vladimir-dudnik should layout flag support dynamic layout or layout with unknown dimensions? Because now it doesn't, and there are no any challenging examples, just straightforward `input1:NCHW,input2:NC`.  And I can replace `<layout>" or "<input1>:<layout1>,<input2>:<layout2>`  from help message with this one.

**Comment by vladimir-dudnik** [comment]
@ivikhrev even if dynamic layout not supported (not sure, we already have [wav2vec](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/public/wav2vec2-base/model.yml) with dynamic shape) but at least an example like `--layout inage_info:0:nc` would be useful for those who will see this for the first time

File: demos/common/python/openvino/model_zoo/model_api/adapters/utils.py [hunk 1]
```diff
@@ -59,17 +59,21 @@ def from_user_layouts(input_names: set, user_layouts: dict):
...
-        Parse layout parameter in format "input0[NCHW],input1[NC]" or "[NCHW]" (applied to all inputs)
+        Parse layout parameter in format "input0:NCHW,input1:NC" or "NCHW" (applied to all inputs)
         '''
         if not layout_regex:
             return None
         inputs_layouts = layout_regex.split(',')
         user_layouts = {}
         for layout in inputs_layouts:
-            if not re.fullmatch(r"[^\[\]]*\[[A-Z]+\]", layout):
+            if not re.fullmatch(r".*\:?[A-Z]+", layout):
```
[the remark this answers is not in the document]
Comment (ivikhrev) [reply demos/common/python/openvino/model_zoo/model_api/adapters/utils.py]: Done

File: demos/common/cpp/utils/src/args_helper.cpp [hunk 2]
```diff
@@ -130,25 +130,23 @@ cv::Size stringToSize(const std::string& str) {
...
-        if (searchStr.empty() || searchStr.front() != ',') {
-            break;
-        }
-        searchStr = searchStr.substr(1);
-        startPos = searchStr.find_first_of('[');
+        searchStr.pop_back();
+        colonPos = searchStr.find_last_of(':');
     }
     if (!searchStr.empty()) {
         throw std::logic_error("Can't parse input layout string: " + layout_string);
```
Comment (Wovchena) [inline demos/common/cpp/utils/src/args_helper.cpp:152]: I suggest `std::invalid_argument`.
Or `RuntimeError` for both languages

File: demos/background_subtraction_demo/python/background_subtraction_demo.py [hunk 3]
```diff
@@ -70,7 +70,7 @@ def build_argparser():
                       help='Optional. Background blur strength (by default with value 0 is not applied).')
     args.add_argument('--layout', type=str, default=None,
                       help='Optional. Model inputs layouts. '
-                           'Format "[<layout>]" or "<input1>[<layout1>],<input2>[<layout2>]" in case of more than one input.'
+                           'Ex. NCHW or input0:NCHW,input1:NC in case of more than one input.'
                            'To define layout you should use only capital letters')
```
Comment (Wovchena) [inline demos/background_subtraction_demo/python/background_subtraction_demo.py:74]: There is no such note for C++. `-d` and `--streams` require the same syntax, but they don't have such note either. I think `To define layout you should use only capital letters` should be removed

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: demos/common/cpp/utils/src/args_helper.cpp (hunk 2 above)
Comment (ivikhrev) [reply demos/common/cpp/utils/src/args_helper.cpp]: Changed to `std::invalid_argument`.

File: demos/background_subtraction_demo/python/background_subtraction_demo.py (hunk 3 above)
Comment (ivikhrev) [reply demos/background_subtraction_demo/python/background_subtraction_demo.py]: Removed.

**Review by Wovchena — APPROVED** [review]

**Review by akorobeinikov — APPROVED** [review]

**Pull request merged**